glitch/analysis/security.py

```diff
@@ -1,15 +1,1107 @@
+        super().__init__(tech)
+
+        if tech == Tech.terraform:
+            self.integrity_policy = SecurityVisitor.TerraformIntegrityPolicy()
+            self.https = SecurityVisitor.TerraformHttpWithoutTls()
+            self.ssl_tls_policy = SecurityVisitor.TerraformSslTlsPolicy()
+            self.dnssec = SecurityVisitor.TerraformDnsWithoutDnssec()
+            self.public_ip = SecurityVisitor.TerraformPublicIp()
+            self.access_control = SecurityVisitor.TerraformAccessControl()
+            self.authentication = SecurityVisitor.TerraformAuthentication()
+            self.missing_encryption = SecurityVisitor.TerraformMissingEncryption()
+            self.firewall_misconfig = SecurityVisitor.TerraformFirewallMisconfig()
+            self.threats_detection = SecurityVisitor.TerraformThreatsDetection()
+            self.weak_password_key_policy = SecurityVisitor.TerraformWeakPasswordKeyPolicy()
+            self.sensitive_iam_action = SecurityVisitor.TerraformSensitiveIAMAction()
+            self.key_management = SecurityVisitor.TerraformKeyManagement()
+            self.network_security_rules = SecurityVisitor.TerraformNetworkSecurityRules()
+            self.permission_iam_policies = SecurityVisitor.TerraformPermissionIAMPolicies()
+            self.logging = SecurityVisitor.TerraformLogging()
+            self.attached_resource = SecurityVisitor.TerraformAttachedResource()
+            self.versioning = SecurityVisitor.TerraformVersioning()
+            self.naming = SecurityVisitor.TerraformNaming()
+            self.replication = SecurityVisitor.TerraformReplication()
+        else:
+            self.integrity_policy = SecurityVisitor.EmptyChecker()
+            self.https = SecurityVisitor.EmptyChecker()
+            self.ssl_tls_policy = SecurityVisitor.EmptyChecker()
+            self.dnssec = SecurityVisitor.EmptyChecker()
+            self.public_ip = SecurityVisitor.EmptyChecker()
+            self.access_control = SecurityVisitor.EmptyChecker()
+            self.authentication = SecurityVisitor.EmptyChecker()
+            self.missing_encryption = SecurityVisitor.EmptyChecker()
+            self.firewall_misconfig = SecurityVisitor.EmptyChecker()
+            self.threats_detection = SecurityVisitor.EmptyChecker()
+            self.weak_password_key_policy = SecurityVisitor.EmptyChecker()
+            self.sensitive_iam_action = SecurityVisitor.EmptyChecker()
+            self.key_management = SecurityVisitor.EmptyChecker()
+            self.network_security_rules = SecurityVisitor.EmptyChecker()
+            self.permission_iam_policies = SecurityVisitor.EmptyChecker()
+            self.logging = SecurityVisitor.EmptyChecker()
+            self.attached_resource = SecurityVisitor.EmptyChecker()
+            self.versioning = SecurityVisitor.EmptyChecker()
+            self.naming = SecurityVisitor.EmptyChecker()
+            self.replication = SecurityVisitor.EmptyChecker()
```

Comment:
We might want to find a way to get a list of checkers dynamically. This would avoid this enumeration in multiple places.